compiler/rustc_mir_build/src/thir/constant.rs

```diff
@@ -8,6 +8,66 @@ use crate::build::parse_float_into_scalar;
 pub(crate) fn lit_to_const<'tcx>(
     tcx: TyCtxt<'tcx>,
     lit_input: LitToConstInput<'tcx>,
+) -> Option<ty::Const<'tcx>> {
+    let LitToConstInput { lit, ty, neg } = lit_input;
+
+    let valtree = match (lit, &ty.kind()) {
+        (ast::LitKind::Str(s, _), ty::Ref(_, inner_ty, _)) if inner_ty.is_str() => {
```

Comment:
this function could ignore the types in the litkinds that don't actually matter?